reservation_units/querysets.py

```diff
@@ -0,0 +1,418 @@
+                is_reservable=False,
+            )
+        )
+
+    if reservation_unit.publish_ends:
+        reservation_unit_closed_time_spans.append(
+            TimeSpanElement(
+                start_datetime=reservation_unit.publish_ends,
+                end_datetime=datetime.max.replace(tzinfo=DEFAULT_TIMEZONE),
+                is_reservable=False,
+            )
+        )
+
+    for application_round in reservation_unit.application_rounds.all():
+        reservation_unit_closed_time_spans.append(
+            TimeSpanElement(
+                start_datetime=datetime.combine(
+                    application_round.reservation_period_begin, time.min, tzinfo=DEFAULT_TIMEZONE
+                ),
+                end_datetime=datetime.combine(
+                    application_round.reservation_period_end, time.min, tzinfo=DEFAULT_TIMEZONE
+                )
+                + timedelta(days=1),
+                is_reservable=False,
+            )
+        )
+
+    return reservation_unit_closed_time_spans
+
+
+def _get_soft_closed_time_spans_for_reservation_unit(reservation_unit: "ReservationUnit") -> list[TimeSpanElement]:
+    now = timezone.localtime()
+    reservation_unit_closed_time_spans: list[TimeSpanElement] = []
+
+    if reservation_unit.reservations_min_days_before:
+        reservation_unit_closed_time_spans.append(
+            TimeSpanElement(
+                start_datetime=datetime.min.replace(tzinfo=DEFAULT_TIMEZONE),
+                end_datetime=now + timedelta(days=reservation_unit.reservations_min_days_before),
+                is_reservable=False,
+            )
+        )
+    if reservation_unit.reservations_max_days_before:
+        reservation_unit_closed_time_spans.append(
+            TimeSpanElement(
+                start_datetime=now + timedelta(days=reservation_unit.reservations_max_days_before),
+                end_datetime=datetime.max.replace(tzinfo=DEFAULT_TIMEZONE),
+                is_reservable=False,
+            )
+        )
+
+    return reservation_unit_closed_time_spans
+
+
+def _get_closed_time_spans_for_time_span(
+    time_span: TimeSpanElement,
+    filter_time_start: time | None,
+    filter_time_end: time | None,
+) -> list[TimeSpanElement]:
+    """
+    Generate a list of closed time spans for a time span based on given filter time values.
+
+    This list will contain at most two time spans for every day between the start and end date of this time span.
+    """
+    closed_time_spans: list[TimeSpanElement] = []
+
+    if not filter_time_start and not filter_time_end:
+        return closed_time_spans
+
+    # Loop through every day between the start and end date of this time span
+    for day in time_span.get_dates_range():
+        # Add closed time spans for the time range outside of the given filter range
+        # e.g. Filter time range is 10:00-14:00, add closed time spans for 00:00-10:00 and 14:00-00:00
+        if filter_time_start:
```

Comment:
I guess here it could be `if filter_time_start and filter_time_start != time.min` to avoid adding zero-length timespans. Same below for `filter_time_end`